rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java

```diff
@@ -93,25 +104,42 @@ public BridgeSupport newInstance(Repository repository, Block executionBlock,
         PeginInstructionsProvider peginInstructionsProvider = new PeginInstructionsProvider();
 
         return new BridgeSupport(
-                bridgeConstants,
-                provider,
-                eventLogger,
-                btcLockSenderProvider,
-                peginInstructionsProvider,
-                repository,
-                executionBlock,
-                btcContext,
-                federationSupport,
-                feePerKbSupport,
-                btcBlockStoreFactory,
-                activations,
-                signatureCache
+            bridgeConstants,
+            provider,
+            eventLogger,
+            btcLockSenderProvider,
+            peginInstructionsProvider,
+            repository,
+            executionBlock,
+            btcContext,
+            federationSupport,
+            feePerKbSupport,
+            whitelistSupport,
+            btcBlockStoreFactory,
+            activations,
+            signatureCache
         );
     }
 
-    private FeePerKbSupport newFeePerKbSupportInstance(StorageAccessor bridgeStorageAccessor, BridgeConstants bridgeConstants) {
+    private FeePerKbSupport getFeePerKbSupportInstance(StorageAccessor bridgeStorageAccessor) {
         FeePerKbConstants feePerKbConstants = bridgeConstants.getFeePerKbConstants();
         FeePerKbStorageProvider feePerKbStorageProvider = new FeePerKbStorageProviderImpl(bridgeStorageAccessor);
+
         return new FeePerKbSupportImpl(feePerKbConstants, feePerKbStorageProvider);
     }
+
+    private WhitelistSupport getWhitelistSupportInstance(StorageAccessor bridgeStorageAccessor, ActivationConfig.ForBlock activations) {
+        WhitelistConstants whitelistConstants = bridgeConstants.getWhitelistConstants();
+        WhitelistStorageProvider whitelistStorageProvider = new WhitelistStorageProviderImpl(
+            whitelistConstants.getBtcParams(),
+            activations,
+            bridgeStorageAccessor
```

Comment:
i just noticed something when seeing this.
When doing the FederationSupport refactor, we had [this](https://github.com/rsksmart/rskj/pull/2330#discussion_r1603951907) discussion about the activations and network parameters being part of the class.

This situation is different because the _activations_ are not part of the `WhitelistSupportImpl`, but the _networkParameters_ are. So probably they should be removed from the `WhitelistStorageProvider` class.

About the activations, I think we should take advantage of the fact that the `BridgeSupport` already has an _activations_ field, and create the supports with it.
That way, all the storage providers would only need the bridge storage accessor to be created, standardizing it.
I find this approach cleaner.

wdyt guys? @marcos-iov @nathanieliov @jeremy-then @apancorb

Reply:
Thanks for the idea! Just implemented it, let me know what you think